### PR TITLE
:sparkles: Task report files

### PR DESCRIPTION
**Pull request by jortel**
Add support for:
- TaskReport attached files.  The addon (and tackle2-addon command) can attach files with command output and/or log files.
- File API to support creating empty files.
- File API to support append content - PATCH /files/:id.
- Addon (adapter) to support attach files to the task report.

Reaper updated to support fields with array of refs.  New tag `ref=[]file`

Example:
```
 - '[CMD] Running: /usr/bin/konveyor-analyzer --provider-settings /addon/opt/settings.json --output-file /addon/report.yaml --no-dependency-rules --rules /addon/rules/rulesets/1/rules --rules /addon/rules/rulesets/24/rules --rules /addon/rules/rulesets/20/rules --label-selector konveyor.io/target=cloud-readiness --dep-label-selector !konveyor.io/dep-source=open-source'
    - '[CMD] /usr/bin/konveyor-analyzer succeeded.'
    - '[CMD] Running: /usr/bin/konveyor-analyzer-dep --provider-settings /addon/opt/settings.json --output-file /addon/deps.yaml'
    - '[CMD] /usr/bin/konveyor-analyzer-dep succeeded.'
    - 'Analysis reported. duration: 102.5784ms'
    - '[TAG] Tagging Application 4.'
    - Facts updated.
    - Done.
attached:
    - id: 989
      name: konveyor-analyzer.output
      activity: 44
    - id: 991
      name: konveyor-analyzer-dep.output
      activity: 50
```

### CODE REVIEW

File: migration/v12/model/core.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+type TaskReport struct {
+	Model
+	Status    string
+	Errors    JSON
+	Total     int
+	Completed int
+	Activity  JSON `gorm:"type:json"`
+	Files     JSON `gorm:"type:json" ref:"[]file"`
```
Comment (jortel) [inline migration/v12/model/core.go:199]: **hint**: Relevant change. ^

**Comment by jortel** [comment]
The API tests pass locally.  No idea why it's failing in he job.

File: api/task.go [hunk 2]
```diff
@@ -558,11 +572,15 @@ func (r *Task) With(m *model.Task) {
 	if m.Errors != nil {
 		_ = json.Unmarshal(m.Errors, &r.Errors)
 	}
+	if m.Errors != nil {
```
Comment (mansam) [inline api/task.go:575]: Looks like accidental duplication here.

File: binding/client.go [hunk 3]
```diff
@@ -422,9 +422,19 @@ func (r *Client) FileGet(path, destination string) (err error) {
...
-func (r *Client) FilePut(path, source string, object interface{}) (err error) {
+func (r *Client) FilePost(path, source string, object interface{}) (err error) {
+	if source == "" {
+		fields := []Field{
+			{
+				Name:   api.FileField,
+				Reader: bytes.NewReader([]byte{}),
+			},
+		}
+		err = r.FileSend(path, http.MethodPut, fields, object)
```
Comment (mansam) [inline binding/client.go:435]: Looks like this should be `http.MethodPost`

File: migration/v12/migrate.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package v12
+
+import (
+	"github.com/jortel/go-utils/logr"
+	"github.com/konveyor/tackle2-hub/migration/v12/model"
+	"gorm.io/gorm"
+)
+
+var log = logr.WithName("migration|v11")
```
Comment (mansam) [inline migration/v12/migrate.go:9]: the log name should be `migration|v12`

**Review by mansam — CHANGES_REQUESTED** [review]
Looks good, just a few comments

File: api/task.go (hunk 2 above)
Comment (jortel) [reply api/task.go:575]: Fixed.

File: binding/client.go (hunk 3 above)
Comment (jortel) [reply binding/client.go:435]: Fixed.

File: migration/v12/migrate.go (hunk 4 above)
Comment (jortel) [reply migration/v12/migrate.go:9]: Fixed.
We may want to correct this in the tooling.

**Review by mansam — APPROVED** [review]
LGTM

**Pull request merged**